### ISSUE TITLE
Remove BCP functionality from here

### ISSUE THREAD

**Issue by webmaster128** [CosmWasm/cosmwasm-js#179]
The package @cosmwasm/bcp was [a successful experiment](https://medium.com/confio/adding-cosmwasm-to-the-neuma-multichain-wallet-ec657d893268) to show token contracts in a multichain wallet. There is no intend to add this functionality into a BCP based wallet like Neuma in production quality. In order to simplify our codebase, we plan the following steps:

1. Refactor @cosmwasm/faucet to work with @cosmwasm/sdk directly and remove the dependency on @cosmwasm/bcp. This implies the removal of smart contract token support from the faucet. It is simplifies to just support bank tokens, which is most important for fees.
2. Remove CosmWasm specifics (ERC20 smart contract token support) from @cosmwasm/bcp
3. Hand over @cosmwasm/bcp to IOV Core (under which target name?)

Mid-term additions

4. Split @cosmwasm/sdk into Cosmos SDK and CosmWasm functionality, such that CosmWasm becomes optional

@davepuchyr Number 2. and 3. is service work for IOV to express our gratitude for being able to build on top of IOV Core and maintain healty code for everyone in the ecosystem. Please folllow this ticket closely and chip in if I get something wrong.

**Comment by ethanfrey** [comment]
This sounds good.

Happy to hand off the @cosmwasm/bcp experiment to iov-core if they want to maintain it, integrate it in their wallet or in any way make real use of it.

And yes, I also agree on refactoring the faucet to just use @cosmwasm/sdk (bank).

**Comment by ethanfrey** [comment]
However, I am not sure about

> Split @cosmwasm/sdk into Cosmos SDK and CosmWasm functionality, such that CosmWasm becomes optional

We are talking about making a new repo to be modular and extensible, pulling in parts of iov-core and cosmwasm-js and others and making sure this is extensible for other modules. I would leave these as is here (working for all out 0.38-based cosmwasm releases) and do the refactoring on the new branch which will have plenty of redesign tasks as well. Unless you think there is a benefit to refactoring here first. (In any case, this is a separate ticket - bcp stuff is a full :+1:)

**Comment by webmaster128** [comment]
As Ethan pointed out correctly, _Split @cosmwasm/sdk into Cosmos SDK and CosmWasm functionality, such that CosmWasm becomes optional_ is independent of this ticket. I created https://github.com/CosmWasm/cosmwasm-js/issues/182 for that. The order of #179 and #182 does not really matter technically.

**Comment by webmaster128** [comment]
The line of PRs #183, #184 and #186 does step 4 and 2 first. So we have no ERC20 tokens on BCP and the faucet after that.

I postponed step 1 and 3 since it is not trivial to refactor the faucet. It required one of those two options:
1. Reimplement the token handling (COMS <-> ucosm) in the faucet
2. Only work on the native unit (e.g. ucosm) and adapt a lot of documentation

**Comment by webmaster128** [comment]
@davepuchyr Under which name would you like to have the BCP Cosmos SDK 0.38 package? `@iov/cosmos-sdk`?

**Comment by davepuchyr** [comment]
> @davepuchyr Under which name would you like to have the BCP Cosmos SDK 0.38 package? `@iov/cosmos-sdk`?

Yes, `@iov/cosmos-sdk`, please.

**Comment by webmaster128** [comment]
Package is prepared. Move depends on #202
